src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAICrafting.java

```diff
@@ -164,19 +169,20 @@ private IAIState queryItems()
 
     /**
      * Check for all items of the required recipe.
+     *
      * @return the next state to go to.
      */
     protected IAIState checkForItems(@NotNull final IRecipeStorage storage)
     {
         final List<ItemStorage> input = storage.getCleanedInput();
-        for(final ItemStorage inputStorage : input)
+        for (final ItemStorage inputStorage : input)
         {
             final Predicate<ItemStack> predicate = stack -> !ItemStackUtils.isEmpty(stack) && new Stack(stack).matches(inputStorage.getItemStack());
             if (InventoryUtils.getItemCountInItemHandler(worker.getInventoryCitizen(), predicate) < inputStorage.getAmount())
             {
                 if (InventoryUtils.hasItemInProvider(getOwnBuilding(), predicate))
                 {
-                    needsCurrently = predicate;
+                    needsCurrently = new Tuple<>(predicate, Constants.STACKSIZE);
```

Comment:
why Constants.STACKSIZE instead of inputStorage.amount?

Reply:
because i experienced that ItemStack sometimes has more than STACKSIZE ( see in pickupMaterial() when it creates the list.)